.github/scripts/manage-images.py

```diff
@@ -0,0 +1,207 @@
+from yandexcloud import SDK, RetryInterceptor, backoff_linear_with_jitter
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s: %(levelname)s: %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+# they won't appear in the list of images
+# but they will be protected from deletion anyway
+PROTECTED_IMAGE_IDS = [
+    "arl2kdkt7j8aorb1ib1h",
+    "arlf5b2mm2k1s4b71u4t",
+    "arlhviu5hj9aj6uqq6l3",
+    "arlpgqq65s5p13srj592",
+    "arl7p6s8d25e8m83kvi3",
+]
+
+
+def status_to_string(image: Image) -> str:
+    return Image.Status.Name(image.status)
+
+
+def convert_size(size_bytes):
+    if size_bytes == 0:
+        return "0 B"
+    size_name = ("B", "KB", "MB", "GB", "TB")
+    i = math.floor(math.log(size_bytes, 1024))
+    p = math.pow(1024, i)
+    s = round(size_bytes / p, 2)
+    return f"{s} {size_name[i]}"
+
+
+def main(
+    sdk: SDK,
+    github_token: str,
+    github_repository: str,
+    new_image_id: str,
+    image_variable_name: str,
+    update_image_id: bool,
+    images_to_keep: int,
+    remove_old_images: bool,
+    folder_id: str,
+) -> None:
+    github = Github(github_token)
+    repo = github.get_repo(github_repository)
+    github_repository_owner, github_repository_name = github_repository.split("/")
+    image_family_name = (
+        f"github-runner-{github_repository_owner}-{github_repository_name}"
+    )
+
+    variable = repo.get_variable(image_variable_name)
+
+    logger.info("%s (old) = %s" % (image_variable_name, variable.value))
+
+    if new_image_id is None:
+        new_image_id = variable.value
+
+    if update_image_id:
+        variable.edit(value=new_image_id)
+        logger.info("%s (new) = %s", image_variable_name, new_image_id)
+    else:
+        logger.info("Would set %s (new) = %s", image_variable_name, new_image_id)
+
+    request = ListImagesRequest(
+        folder_id=folder_id,
+        order_by="created_at desc",
+        filter=f"family IN ('{image_family_name}') AND status = 'READY'",
+    )
+
+    client = sdk.client(ImageServiceStub)
+    response = client.List(request)
+    candidate_images = []
+    for image in response.images:
+        status = status_to_string(image)
+        created_at = datetime.fromtimestamp(image.created_at.seconds, tz=timezone.utc)
+        storage_size = convert_size(image.storage_size)
+        min_disk_size = convert_size(image.min_disk_size)
+        prefix = " (PROTECTED)"
+        postfix = ""
+        if (
+            image.id not in PROTECTED_IMAGE_IDS
+            and image.family == image_family_name  # noqa: W503
+            and image.id != new_image_id  # noqa: W503
+        ):
+            prefix = ""
+            if image.id == new_image_id:
+                postfix = " (NEW)"
+            if image.family == image_family_name:
+                postfix += " (SELECTED)"
+            candidate_images.append(image)
+
+        logger.info(
+            "Found image%s: %s %s %s %s %s %s %s %s",
```

Comment:
In logging you can name parameters via %(name)s

Reply:
To do that you need to pass huge dict with keys and all, and it doesn't look good and looks excsessive.